Tests/Rules/ReviewUnusedParameter.tests.ps1

```diff
@@ -0,0 +1,68 @@
+Describe "ReviewUnusedParameter" {
+    BeforeAll {
+        $RuleName = 'PSReviewUnusedParameter'
+        $RuleSeverity = "Warning"
+    }
+
+    Context "When there are violations" {
+        $Cases = @(
+            @{
+                ScriptDefinition   = 'function BadFunc1 { param ($Param1, $Param2) $Param1}'
+                Name               = "function with 1 unused parameter"
+                NumberOfViolations = 1
+            }
+            @{
+                ScriptDefinition   = 'function BadFunc1 { param ($Param1, $Param2) }'
+                Name               = "function with 2 unused parameters"
+                NumberOfViolations = 2
+            }
+            @{
+                ScriptDefinition   = '{ param ($Param1) }'
+                Name               = "scriptblock with unused parameter"
+                NumberOfViolations = 1
+            }
+            @{
+                ScriptDefinition   = '{ param ($Param1) }; { $Param1 }'
+                Name               = "parameter with same name in different scope"
+                NumberOfViolations = 1
+            }
+        )
```

Comment:
```suggestion
        )

```

Reply:
fixed in 938a2835ac8d3c8e5470642d923c3b85476b303c